File: blepro/src/main/java/com/lepu/blepro/ext/bp2w/Bp2wConfig.java
package com.lepu.blepro.ext.bp2w;

public class Bp2wConfig {
    private boolean soundOn;            // 心电音开关
    private int avgMeasureMode;         // 0：x3模式关闭 1：x3模式开启（时间间隔30s） 2：时间间隔60s 3：时间间隔90s 4：时间间隔120s

    public boolean isSoundOn() {
        return soundOn;
    }

    public void setSoundOn(boolean soundOn) {
        this.soundOn = soundOn;
    }

    public int getAvgMeasureMode() {
        return avgMeasureMode;
    }

    public void setAvgMeasureMode(int avgMeasureMode) {
        this.avgMeasureMode = avgMeasureMode;
    }

    @Override
    public String toString() {
        return "Bp2wConfig{" +
                "soundOn=" + soundOn +
                ", avgMeasureMode=" + avgMeasureMode +
                '}';
    }
}

File: blepro/src/main/java/com/lepu/blepro/ble/data/Bp2wConfig.kt
package com.lepu.blepro.ble.data

import com.lepu.blepro.utils.toUInt

class Bp2wConfig {

    var prevCalibZero: Int
    var lastCalibZero: Int
    var calibSlope: Int
    var slopePressure: Int
    var calibTicks: Int
    var sleepTicks: Int
    var bpTestTargetPressure: Int
    var beepSwitch: Int
    var avgMeasureMode: Int
    var volume: Int

    constructor(bytes: ByteArray) {
        var index = 0
        prevCalibZero = toUInt(bytes.copyOfRange(index, index+4))
        index += 4
        lastCalibZero = toUInt(bytes.copyOfRange(index, index+4))
        index += 4
        calibSlope = toUInt(bytes.copyOfRange(index, index+4))
        index += 4
        slopePressure = toUInt(bytes.copyOfRange(index, index+2))
        index += 2
        calibTicks = toUInt(bytes.copyOfRange(index, index+4))
        index += 4
        sleepTicks = toUInt(bytes.copyOfRange(index, index+4))
        index += 4
        bpTestTargetPressure = toUInt(bytes.copyOfRange(index, index+2))
        index += 2
        beepSwitch = toUInt(bytes.copyOfRange(index, index+1))
        index++
        avgMeasureMode = toUInt(bytes.copyOfRange(index, index+1))
        index++
        volume = toUInt(bytes.copyOfRange(index, index+1))
    }

    override fun toString(): String {
        return """
            Bp2wConfig
            prevCalibZero : $prevCalibZero
            lastCalibZero : $lastCalibZero
            calibSlope : $calibSlope
            slopePressure : $slopePressure
            calibTicks : $calibTicks
            sleepTicks : $sleepTicks
            bpTestTargetPressure : $bpTestTargetPressure
            beepSwitch : $beepSwitch
            avgMeasureMode : $avgMeasureMode
            volume : $volume
        """.trimIndent()
    }
}